docs/howto/features/imagebuilding.md

```diff
@@ -0,0 +1,153 @@
+```
+
+2. Since `jupyterhub-fancy-profiles` adds on to the [`profileList` feature of `KubeSpawner`](https://jupyterhub-kubespawner.readthedocs.io/en/latest/spawner.html#kubespawner.KubeSpawner.profile_list), we need to configure a profile list here as well. Edit and update the following configuration per the community requesting the hub needs.
+
+```yaml
+singleuser:
+  profileList:
+    - display_name: "Only Profile Available, this info is not shown in the UI"
+      slug: only-choice
+      profile_options:
+        image:
+          display_name: Image
+          unlisted_choice:
+            enabled: True
+            display_name: "Custom image"
+            validation_regex: "^.+:.+$"
+            validation_message: "Must be a publicly available docker image, of form <image-name>:<tag>"
+            display_name_in_choices: "Specify an existing docker image"
+            description_in_choices: "Use a pre-existing docker image from a public docker registry (dockerhub, quay, etc)"
+            kubespawner_override:
+              image: "{value}"
+          choices:
+            pangeo:
+              display_name: Pangeo Notebook Image
+              description: "Python image with scientific, dask and geospatial tools"
+              kubespawner_override:
+                image: pangeo/pangeo-notebook:2023.09.11
+            scipy:
+              display_name: Jupyter SciPy Notebook
+              slug: scipy
+              kubespawner_override:
+                image: jupyter/scipy-notebook:2023-06-26
+```
+
+## The `binderhub-service` chart
+
+We will use the [binderhub-service](https://github.com/2i2c-org/binderhub-service/) Helm chart to run BinderHub, the Python software, as a standalone service to build and push images with [repo2docker](https://github.com/jupyterhub/repo2docker), next to JupyterHub.
+
+### How it works
+
+The `binderhub-service` installation starts a `docker-api` pod on each of the user nodes via the following [DaemonSet definition](https://github.com/2i2c-org/binderhub-service/blob/main/binderhub-service/templates/docker-api/daemonset.yaml).
+
+The `docker-api` pod setups and starts the [dockerd](https://docs.docker.com/engine/reference/commandline/dockerd/) daemon, that will then be accessible via a mounted unix socket, by the `build pods`.
+
+The `build pods` are created as a result of an image build request, and they must run on the same node as the builder pods to make use of the docker daemon. These pods mount a k8s Secret with the docker config file holding the necessary registry credentials.
+
+### How to set it up
+
+1. Setup a registry and safely store the credentials to push to it
+
+  - If the hub will be deployed on a GCP cluster, follow the steps 4 to 8 from the [binderhub-service installation guide](https://github.com/2i2c-org/binderhub-service?tab=readme-ov-file#installation). When you're done, you should have a gcr.io registry setup and the appropriate credentials to push images to it.
```

Comment:
In our infrastructure for GCP, we actually have https://github.com/2i2c-org/infrastructure/blob/master/terraform/gcp/registry.tf in terraform that sets this up.

It has to be enabled in terraform [like this](https://github.com/2i2c-org/infrastructure/blob/a18937fbe413428b98515d0b5c713aacb5a0d56b/terraform/gcp/projects/pilot-hubs.tfvars#L83). One entry per registry we want to have.

And then, you can get the password for the credentials by running `terraform output  registry_sa_keys`. The username is `_json_key` ([source](https://cloud.google.com/artifact-registry/docs/docker/authentication#json-key)), and the password is the service account JSON that `terraform output` provides.

This is what we have for the imagebuilding-demo hub: https://github.com/2i2c-org/infrastructure/blob/a18937fbe413428b98515d0b5c713aacb5a0d56b/config/clusters/2i2c/imagebuilding-demo.values.yaml#L177.

So I suggest we split this up into its own subsection on 'setting up the image registry'. *That* can have two subsections - one for GCP (with terraform), and one for everything else (with quay.io). As we add native support (via terraform) for other cloud providers, we can add those here.